### ISSUE TITLE
Crash in reader detail view

### ISSUE THREAD

**Issue by roundhill** [wordpress-mobile/WordPress-iOS#3027]
I can repro every time I load a certain post in the reader:

http://exploratorius.us/2014/12/31/my-best-images-from-2014/

That post is in Freshly Pressed. It hangs for a long while, then crashes.

iPad mini retina, iOS 8.1

**Comment by aerych** [comment]
Thanks Dan.  I haven't been able to reproduce on two devices or in the simulator.  Could you send me the crash log if its handy?

**Comment by roundhill** [comment]
Sure, strangely it says 'Unknown' in the logs listing:

https://cloudup.com/ipSEIuHMuH8

Could it be running out of memory? That's a ton of images in that post! Also, the WP Internal app doesn't prompt me to send a crash report... not sure if that helps isolate it.

**Comment by aerych** [comment]
I'm liking low memory as the cause but we'll see.   Thanks for the log!!

**Comment by aerych** [comment]
`WordPress <10951ca3aa913fe89193c465988672c3>        29.290      180506                0           -         59204  100  [per-process-limit]  (frontmost) (resume)`

Yah. I'd say low memory seems like a good candidate :D

**Comment by koke** [comment]
I've been able to reproduce this on an iPad mini after some browsing in Safari. Definitely a low-memory kill and not a crash.

I profiled loading the detail view and there's a spike in memory usage. From what I understand, the reader detail processes the content, and proceeds to load all the images with `WPImageSource`, which uses a concurrent queue for processing with no limit. This means that when many of the image requests complete, they're all processed at once causing a spike in memory usage.

**Comment by kurzee** [comment]
@aerych quick note, we just tested this one and it's definitely still crashing on an iPad mini and probably older devices. Moving it off being an iPad specific project.

**Comment by aerych** [comment]
Totally not an ipad only issue.  I expect its just due to how many images we the detail tries to hold in memory at once.  I've an experimental branch in the works that rebuilds the detail as a tableview as a way of limiting what we hold in memory.

**Comment by bummytime** [comment]
Just a friendly update - still an issue with the latest `develop` build on an iPhone 5 running 10.3.3. 

@aerych Do you have any updated thoughts on this? Does that experimental branch still exist? 😄

**Comment by aerych** [comment]
> Does that experimental branch still exist

Yes, but I didn't get very far with it.  I haven't been able to loop back to it either 😢   I imagine its fairly out of date at this point due to other changes. 
I was tracking in https://github.com/wordpress-mobile/WordPress-iOS/issues/7055  which should resolve this issue (if I can ever get back to it --- or if someone else want's to take the reigns).

**Comment by SiobhyB** [comment]
This came up again after version 9.7.2 of the app froze and crashed for me while viewing a long post with lots of images in the Reader. I use an iPad Mini 2 and am on the latest version of iOS (11.3). Discussion at p1524132675000395-slack-cellular-happiness.

**Comment by jkmassel** [comment]
Identified a situation that I think is the same bug in version 10.7.

**Comment by designsimply** [comment]
I tried testing this today using WP Internal 12.4.0.20190520 on iPad Mini 4 (WiFi) (iPad5,1) iOS 12.2 by liking a post with ~85 individual images ([post tested](https://walleeestravels.com/2019/04/23/phu-quoc-vietnam/)) embedded and then going to Reader > My Likes in the app and viewing that post—I was able to view the post normally. 

http://exploratorius.us/2014/12/31/my-best-images-from-2014/ now redirects to http://exploratorius.us/ so I am not sure how many images that post had and that would be good to note for any future reports.

If it does not come up again by the time the next iOS Groundskeeping rolls around, I will move it to the `Prioritized iOS` list for one more test and recommending it as a candidate to close if another tester is unable to replicate the problem.
